src/pivot-table/hooks/use-column-width.ts

```diff
@@ -118,11 +123,11 @@ export default function useColumnWidth(
       return width;
     });
 
-    const leftGridMaxWidth = rect.width * LEFT_GRID_MAX_WIDTH_RATIO;
-    if (sumOfWidths < leftGridMaxWidth) return widths;
-
-    const multiplier = leftGridMaxWidth / sumOfWidths;
-    return widths.map((w) => w * multiplier);
+    return {
+      columnWidths,
+      leftGridWidth: Math.min(rect.width * LEFT_GRID_MAX_WIDTH_RATIO, sumOfWidths),
+      getLeftGridColumnWidth: (idx) => columnWidths[idx],
```

Comment:
I don't really see any point in having this function anymore. Might as well just consume the `columnWidths` array. But no need to change it in this PR.

Reply:
I removed it